components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacetImpl.java

```diff
@@ -91,13 +94,14 @@ protected void put(final EntityId componentId) {
       additional.put(P_REPOSITORY_NAME, getRepository().getName());
       Component component;
       List<Asset> assets;
-      try (StorageTx tx = facet(StorageFacet.class).openTx()) {
-        component = tx.findComponent(componentId, tx.getBucket());
-        if (component == null) {
-          return;
-        }
-        assets = Lists.newArrayList(tx.browseAssets(component));
+
+      final StorageTx tx = UnitOfWork.currentTransaction();
+      component = tx.findComponent(componentId, tx.getBucket());
+      if (component == null) {
+        return;
       }
+      assets = Lists.newArrayList(tx.browseAssets(component));
```

Comment:
agree, but the change here is just shifting code left - how about a separate PR removing all use of newArrayList, etc?